eval/value.go

```diff
@@ -79,7 +79,8 @@ func newLiteralValueExpr(ctx *hcl.EvalContext, exp hcl.Expression) hclsyntax.Exp
 		}
 
 		typ := c.Type()
-		if typ == cty.Bool && c.False() {
+		if typ == cty.Bool && c.False() ||
+			typ == cty.String && c.AsString() == "" {
```

Comment:
According to documentation, for `.AsString()` not to panic, the value must be
> a non-null, non-unknown cty.String value

Add `&& !c.IsNull() && c.IsKnown() ...`?

Reply:
`c.IsNull` is always `false` here, not sure about "known", though.